src/Handlers/BookingsStateStatusesHandler.php

```diff
@@ -217,14 +213,28 @@ protected function _getScreenStatuses($key, $defaultStatuses = [])
 
         $screenOptions = get_user_option($key, $user->ID);
         if (!$screenOptions) {
-            return $this->_normalizeArray($defaultStatuses);
+            return $this->_getVisibleStatuses($defaultStatuses);
         }
-        $screenOptions = json_decode($screenOptions);
 
-        $statuses = $this->_trigger('eddbk_bookings_visible_statuses', [
-            'statuses' => $screenOptions,
-        ])->getParam('statuses');
+        $screenOptions = json_decode($screenOptions);
```

Comment:
Are there going to me multiple cases where `_getScreenStatuses()` needs to be called with different keys? If not, why not abstract retrieval of the user options, or better - the whole process of retrieving screen options?